sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,233 @@
+import { addDays } from "@sims/utilities";
+
+describe("ApplicationStudentsController(e2e)-getCompletedApplicationDetails", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let applicationRepo: Repository<Application>;
+  let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+  let studentScholasticStandingRepo: Repository<StudentScholasticStanding>;
+  let studentAppealRepo: Repository<StudentAppeal>;
+  let submittedByInstitutionUser: User;
+  let student: Student;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    const userRepo = dataSource.getRepository(User);
+    applicationRepo = dataSource.getRepository(Application);
+    disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+    studentScholasticStandingRepo = dataSource.getRepository(
+      StudentScholasticStanding,
+    );
+    studentAppealRepo = dataSource.getRepository(StudentAppeal);
+    submittedByInstitutionUser = await userRepo.save(createFakeUser());
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should throw NotFoundException when the application is not associated with the authenticated student.", async () => {
+    // Arrange
+    // The application will be generated with a fake user.
+    const application = await saveFakeApplicationDisbursements(appDataSource);
+    const endpoint = `/students/application/${application.id}/completed`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: 404,
+        message: "Application not found or not on Completed status.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should throw NotFoundException when the application is not in 'Completed' status.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      { student },
+      { createSecondDisbursement: true },
+    );
+    const endpoint = `/students/application/${application.id}/completed`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: 404,
+        message: "Application not found or not on Completed status.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should get application details when application is in 'Completed' status.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
```

Comment:
mm, instead of getting a fake disbursement application and changing the data in the test, can't we create a fakecompletedAppliaction? is there any reason for it?

Reply:
When we need an application with disbursements we will need: `createFakeApplication`, `createFakeStudentAssessment`, `createFakeDisbursementSchedule`, `createFakeEducationProgramOffering`, `createFakeUser`, `createFakeStudent`, etc., and then have all of them saved properly. When this method was introduced in the past for COE (and now renamed/moved for disbursements), the idea was to concentrate all the complexity of calling the previously mentioned `createFake` methods and coordinate the way that they must be saved.
Does it answer? If not maybe I am not getting the point and we can have a chat. Please let me know.

Reply:
I guess that the answer is the same from https://github.com/bcgov/SIMS/pull/1832#discussion_r1148234270